Lib/functools.py

```diff
@@ -949,6 +952,13 @@ def _method(*args, **kwargs):
         _method.__isabstractmethod__ = self.__isabstractmethod__
         _method.register = self.register
         update_wrapper(_method, self.func)
+
+        if self.attrname is not None:
+            try:
+                obj.__dict__[self.attrname] = _method
+            except AttributeError:
+                pass  # not all objects have __dict__ (e.g. classes with __slots__)
```

Comment:
Since we now never set it to `None` in `__init__`, it can never be `None`, so there's no need for this check:

```suggestion
        try:
            obj.__dict__[self.attrname] = _method
        except AttributeError:
            pass  # not all objects have __dict__ (e.g. classes with __slots__)
```